src/main/java/com/gildedrose/AgedBrie.java

```diff
@@ -1,27 +1,29 @@
 package com.gildedrose;
 
-public class AgedBrie {
+public class AgedBrie implements Ageable {
     private int qualityIncrementation;
+    private final Item item;
 
-    public AgedBrie() {
+    public AgedBrie(Item item) {
+        this.item = item;
         this.qualityIncrementation = 1;
     }
 
-    public void age(Item item) {
-        item.quality = updateQuality(item.quality, item.sellIn);
-        item.sellIn = updateSellIn(item.sellIn);
+    public void age() {
+        item.quality = updateQuality();
+        item.sellIn = updateSellIn();
     }
 
-    private int updateQuality(int quality, int sellIn) {
-        if (sellIn <= 0) {
+    private int updateQuality() {
+        if (item.sellIn <= 0) {
             this.qualityIncrementation = 2;
         }
-        quality += this.qualityIncrementation;
-        return Math.min(quality, 50);
+        item.quality += this.qualityIncrementation;
```

Comment:
While "incrementation" is an interesting word, I think the word you're looking for is "coefficient". Coefficient is a domain specific term in math and computer science and will be understood by more readers of the program.

Reply:
Also "increment" does not make sense when the quality goes negative like for backstage passes.

Reply:
I like coefficient better, will be replacing the"incrementation" and "degradation" naming conventions with it, thank you 😄

Reply:
Addressed in [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/19/commits/af97ebb60d4bf4c4ce87f19a77b589f808f1f30c)